src/shared/localShadowRepo.ts

```diff
@@ -59,7 +61,9 @@ export class ShadowRepo {
     this.projectPath = options.projectPath;
     this.packageDirs = options.packageDirs;
     this.isWindows = os.type() === 'Windows_NT';
-    this.maxFileAdd = this.isWindows ? 8000 : 15000;
+
+    const batchSize = env.getNumber('SFDX_SOURCE_TRACKING_BATCH_SIZE');
+    this.maxFileAdd = batchSize ? batchSize : this.isWindows ? 8000 : 15000;
```

Comment:
rather than nested ternary, you could do something like

```suggestion
    this.maxFileAdd = batchSize ??  this.isWindows ? 8000 : 15000;
```

or just
```
this.maxFileAdd = env.getNumber('SFDX_SOURCE_TRACKING_BATCH_SIZE') ?? this.isWindows ? 8000 : 15000
```

or use the "default" 2nd prop of env,
this.maxFileAdd = env.getNumber('SFDX_SOURCE_TRACKING_BATCH_SIZE',  this.isWindows ? 8000 : 15000)

Reply:
Yea, I generally don't like nested ternaries but I like the combination of a nullish coalescing and a ternary less. Mainly because it is not immediately obvious what is being compared. The code you suggested would not work correctly, it would compare the nullish first and use that result for the ternary. If you want to combine the two operators, you'd need to wrap your ternary in parentheses. 

```js
// resolves 'tern-true' 👎
console.log('foo' ?? true ? 'tern-true' : 'tern-false');

// resolves 'foo'
console.log('foo' ?? (true ? 'tern-true' : 'tern-false'));

// resolves ""
console.log('' ?? (true ? 'tern-true' : 'tern-false'));

// resolves "tern-true"
console.log(undefined ?? (true ? 'tern-true' : 'tern-false'));

// resolves "tern-false"
console.log(undefined ?? (false ? 'tern-true' : 'tern-false'));
```
https://jsfiddle.net/hxjzaq5c/1/

I do like the idea of using the default for `getNumber`, I will switch to that. Thanks for the suggestion